src/js/pages/catalog/PackageDetailTab.js

```diff
@@ -78,15 +84,73 @@ class PackageDetailTab extends mixin(StoreMixin) {
     });
   }
 
-  componentDidMount() {
-    super.componentDidMount(...arguments);
+  isSelectedVersionLoading() {
+    const { version } = this.props.location.query;
+    const cosmosPackage = CosmosPackagesStore.getPackageDetails();
+
+    return cosmosPackage.getVersion() !== version;
+  }
 
+  retrievePackageInfo() {
     const { packageName } = this.props.params;
     const { version } = this.props.location.query;
+    const packageVersions = CosmosPackagesStore.getPackageVersions(packageName);
+
+    // fetch package versions available only if not cached
+    if (packageVersions == null) {
+      CosmosPackagesStore.fetchPackageVersions(packageName);
+    }
+
     // Fetch package description
     CosmosPackagesStore.fetchPackageDescription(packageName, version);
   }
 
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+
+    this.retrievePackageInfo();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const isLoadingVersions =
+      CosmosPackagesStore.getPackageVersions(nextProps.params.packageName) ==
+      null;
+
+    this.setState({ isLoadingVersions });
```

Comment:
If possible please create a general `isLoading` helper that check both the versions as well as the details.

Reply:
We don't need to block the page for both requests when we only need to wait for one to render 99% of the contents.